lib/upi-consumer.coffee

```diff
@@ -263,7 +279,7 @@ class UPIConsumer
         else
           throw new Error('Got neither type nor info')
       text:
-        text: "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
+        text: (Object.keys(doc).join(", ") + "\nBLERG 0: " + doc[Object.keys(doc)[0]] + "\nBLERG 1: " + doc[Object.keys(doc)[1]] + "\nBLERG 1 keys: " + Object.keys(doc[Object.keys(doc)[1]]) + "\nBLERG 1[0]: " + doc[Object.keys(doc)[1]]["text"] + "--(Karl was here)\n") + "#{if type?.text?.text then ':: '+type.text.text+'\n' else ''}#{info?.text?.text ? ''}"
```

Comment:
If you're trying to debug it, you might consider using `JSON.stringify(doc)` instead of this long-winded construction.

That said, I have two thoughts on this whole thing. First, I would think that since this has virtually nothing to do with ghc-mod, this would be better-suited to be provided as a separate package, unless it eventually does (see below). 

Second, there's some work wrt haddock happening on ghc-mod, which you might want to take a look at: https://github.com/DanielG/ghc-mod/pull/810